### PR TITLE
Estimate anti join selectivity by least-selective equi-join-clause

### PR DESCRIPTION
**Pull request by sopel39**
Making it consistent with how inner join works

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/cost/JoinStatsRule.java [hunk 1]
```diff
@@ -187,59 +187,76 @@ private static double firstNonNaN(double... values)
...
-                return UNKNOWN_STATS;
+                // right stats row count is NaN
+                return new AntiJoinStats(UNKNOWN_STATS, ImmutableSet.of());
             }
         }
 
-        return outputStats;
+        // heuristics: select the most selective criteria for anti join clause
+        return criteria.stream()
+                .map(clause -> {
```
Comment (rschlussel-zz) [inline presto-main/src/main/java/com/facebook/presto/cost/JoinStatsRule.java]: this function is kind of long for an anonymous function.  Can you make it not anoymous so it's easier to see the flow?

**Review by rschlussel-zz — COMMENTED** [review]
1. For inner join stats we filter by unknown for the remaining  predicates.  Can we do that here too?
2. Can you add a test?

**Comment by sopel39** [comment]
> For inner join stats we filter by unknown for the remaining predicates. Can we do that here too?

Yeah, I guess I could multiply anti join stats by `1/UNKNOWN_FILTER_COEFFICIENT`.
It's not obvious though how to modify other clauses column stats since it's enough that driving clause fails for row to be part of anti join

**Comment by rschlussel-zz** [comment]
Interesting.  I guess keep the column stats the same and only update the row count.  I think it's still worth updating the rowcount, at least because we want outer join computations to be somewhat coherent, and that's basically innerjoin + antijoin.

**Comment by rschlussel-zz** [comment]
Also, I'm on PTO tomorrow.  No need to wait for me to merge.

File: presto-main/src/main/java/com/facebook/presto/cost/JoinStatsRule.java [hunk 2]
```diff
@@ -293,24 +325,33 @@ private PlanNodeStatsEstimate crossJoinStats(JoinNode node, PlanNodeStatsEstimat
...
-    private Set<Symbol> getRightJoinSymbols(JoinNode node)
+    @VisibleForTesting
+    static class AntiJoinStats
     {
-        return node.getCriteria().stream()
-                .map(EquiJoinClause::getRight)
-                .collect(toImmutableSet());
-    }
+        private final PlanNodeStatsEstimate stats;
+        private final Set<Symbol> selectedClauseSymbols;
```
Comment (losipiuk) [inline presto-main/src/main/java/com/facebook/presto/cost/JoinStatsRule.java]: `Optional selectedClauseSymbol`

**Review by losipiuk — APPROVED** [review]

**Pull request merged**